solr/core/src/java/org/apache/solr/schema/WrappedValueSource.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.schema;
+
+import java.io.IOException;
+import java.util.Map;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.queries.function.FunctionValues;
+import org.apache.lucene.queries.function.ValueSource;
+import org.apache.lucene.search.DoubleValuesSource;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.LongValuesSource;
+import org.apache.lucene.search.SortField;
+
+/**
+ * Wraps a ValueSource that derives values directly from a particular SchemaField. This allows a
+ * relatively clean way to set missingValue on ValueSource.getSortField, and also clearly marks
+ * SortFields that are directly associated with a SchemaField (e.g., for purposes of
+ * marshaling/unmarshaling sort values).
+ */
+public class WrappedValueSource extends ValueSource {
```

Comment:
Excuse me. I can't find usages for this class.

Reply:
True; a side-effect of my hasty migration from the old PR, I think? I think it can just be removed, not sure what purpose it ever served, if any.